File: baekjoon/11723.py
#!/usr/bin/env python3

ans = ""
group = [0 for i in range(21)]

M = int(input())
for m in range(M):
    cmd = input().split(" ")
    if cmd[0] == "add":
        group[int(cmd[1])] = 1
    elif cmd[0] == "remove":
        group[int(cmd[1])] = 0
    elif cmd[0] == "check":
        ans += str(group[int(cmd[1])]) + "\n"
    elif cmd[0] == "toggle":
        group[int(cmd[1])] = int(not group[int(cmd[1])])
    elif cmd[0] == "all":
        for n in range(1, 21):
            group[n] = 1
    else:
        for n in range(1, 21):
            group[n] = 0

print(ans)

File: baekjoon/11723.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <cstdlib>
#include <cstdio>
#include <string>
#include <vector>
#include <random>
#include <algorithm>
#include <cmath>
#include <map>
#include <set>
#include <stack>
#include <queue>
#include <deque>

int main()
{
	std::ios::sync_with_stdio(false);
	std::cin.tie(NULL);
	std::cout.tie(NULL);

	int N; std::cin >> N;

	std::vector<int> v(21, 0);
	while(N--) {
		std::string cmd;
		std::cin >> cmd;

		int x;
		if(cmd != "all" && cmd != "empty")
			std::cin >> x;
		
		if(cmd == "add") v[x] = 1;
		else if(cmd == "remove") v[x] = 0;
		else if(cmd == "toggle") v[x] = !v[x];
		else if(cmd == "check")
			std::cout << v[x] << "\n";
		else if(cmd == "all")
			for(int i = 1; i <= 20; i++) v[i] = 1;
		else
			for(int i = 1; i <= 20; i++) v[i] = 0;
	}

	return 0;
}
